File: Person/Person.ts
///<reference path='./Scripts/typings/moment/moment.d.ts' />

export module Person{
	export module TypeScript {
		export class Person {
			private _birthDate: Date;
			private _name: string;

			constructor(birthDate: Date, name: string) {
				this._birthDate = birthDate;
				this._name = name;
			}

			public toString(): string {
				return this._name + ", " + this._birthDate;
			}

			public get birthDate(): Date {
				return this._birthDate;
			}

			public get name(): string {
				return this._name;
			}
		}
	}
}

File: Person.NET/Person.cs
using NodaTime;

namespace People.NET
{
    public sealed class Person
    {
		 public Person(LocalDate birthDate, string name)
		 {
			 this.BirthDate = birthDate;
			 this.Name = name;
		 }

		 public override string ToString()
		 {
			 return this.Name + " " + this.BirthDate;
		 }

		 public LocalDate BirthDate { get; private set; }
		 public string Name { get; private set; }
	 }
}
